jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -594,6 +595,8 @@ private boolean parseRequestBuffer()
         if (LOG.isDebugEnabled())
             LOG.debug("{} parse {}", this, _retainableByteBuffer);
 
+        if (_parser.isTerminated())
+            throw new RuntimeIOException("Parser is terminated");
```

Comment:
@lachlan-roberts @sbordet I think this is causing #12212
Can you recall why this was necessary?

Reply:
@gregw this was added in d4682f4b85c3e66981bb0544b795db185f8909f9, with the commit description "Fix testMaxRequest size test in MultiPartServletTest".

I re-run those tests, and the changes #12213 pass those tests, so I think either was not necessary (maybe just a zealous sanity check), or it is not anymore.